nodegsettings.cc

```diff
@@ -0,0 +1,385 @@
+			will wrong results on some architectures
+			as the uint max limit can be different.
+			As I understand it, v8 will try to cast
+			the guint value to the achitecture specific
+			uint value. If the max uint value is lower
+			than the one of guint, the value can be corrupted
+			and can (as I experienced it) be set to -1
+			To keep v8 from casting the guint
+			value to a achitecutre dependent lower
+			limited unit
+			we use the more general 'Number' class
+			that supports long values.
+		*/
+		return scope.Close(Number::New(gunit32_value));
+	}
+	else if (g_variant_type_equal(type,G_VARIANT_TYPE_STRING)) {
+		return scope.Close(String::New(g_variant_get_string(variant,NULL)));
+	}
+	else if (g_variant_type_equal(type,G_VARIANT_TYPE_STRING_ARRAY)) {
+		gsize length;
+		const gchar **elems = g_variant_get_strv(variant,&length);
+
+		HandleScope scope;
+		Local<Array> nodes = Array::New();
+
+		for (unsigned int i = 0; i < length; ++i) {
+			nodes->Set(i, String::New(elems[i]));
+		}
+
+		return scope.Close(nodes);
+	}
+	else if (g_variant_type_equal(type, X_G_VARIANT_TYPE_STRING_TUPLE_ARRAY)) {
+		Local<Array> tuple_array = Array::New();
+
+		GVariantIter *iter;
+		gchar *str1;
+		gchar *str2;
+		unsigned int i = 0;
+
+		g_variant_get (variant, "a(ss)", &iter);
+		while (g_variant_iter_loop (iter, "(ss)", &str1, &str2)) {
+			Local<Array> tuple = Array::New();
+			tuple->Set(0, String::New(str1));
+			tuple->Set(1, String::New(str2));
+			tuple_array->Set(i, tuple);
+			i++;
+		}
+		g_variant_iter_free (iter);
+
+		return scope.Close(tuple_array);
+	}
+	else if (g_variant_type_equal(type,G_VARIANT_TYPE_BOOLEAN)) {
+		return scope.Close(Boolean::New(g_variant_get_boolean(variant)));
+	}
+	else {
+		g_print("The type is %s\n", g_variant_type_peek_string(type));
+		ThrowException(Exception::Error(String::New("Need to implement reading that value type")));
+		return scope.Close(Undefined());
+	}
+}
+
+/* Takes schemaId string */
+Handle<Value> schema_exists(const Arguments& args) {
+	HandleScope            scope;
+	char                  *schema_id     = v8_value_to_utf8_string(args[0]);
+	GSettingsSchemaSource *schema_source = g_settings_schema_source_get_default();
+
+	if ( ! schema_source ) {
+		ThrowException(Exception::Error(String::New("No schema sources available!")));
+		return scope.Close(Undefined());
+	}
+
+	GSettingsSchema * schema =
+		g_settings_schema_source_lookup (schema_source,
+		                                 schema_id,
+		                                 FALSE);
+
+	if ( ! schema ) {
+		return scope.Close(Boolean::New(FALSE));
+	}
+
+	return scope.Close(Boolean::New(TRUE));
+}
+
+/* Takes schemaId, key, and value */
+Handle<Value> set_gsetting(const Arguments& args) {
+	HandleScope         scope;
+	Local<Value>        value_obj = args[2];
+	GSettings          *settings;
+	GVariant           *variant;
+	const GVariantType *type;
+	bool                success = false;
+	const char         *schemaId;
```

Comment:
Safest to use a std::string here as per comment above, in order to avoid leaking memory